llama.go

```diff
@@ -23,7 +25,14 @@ func New(model string, opts ...ModelOption) (*LLama, error) {
 		return nil, fmt.Errorf("failed loading model")
 	}
 
-	return &LLama{state: result}, nil
+	ll := &LLama{state: result}
+
+	// set a finalizer to remove any callbacks when the struct is reclaimed by the garbage collector.
+	runtime.SetFinalizer(ll, func(ll *LLama) {
```

Comment:
I have not had much success with this pattern in Go. Not being able to control when the finalizer kicks in can sometimes result in unexpected behavior..

In GoCV we require the caller manually release the C memory. To help prevent leaks we use a custom profiler used when running the unit tests.

Anyhow, just mentioning it in case this comes up later.